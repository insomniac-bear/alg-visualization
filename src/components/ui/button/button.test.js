import renderer from "react-test-renderer";
import { Button } from "./button";

const mockTest = "Test text";

describe("Checking Button component with snapshot", () => {
  it("Button should be render with text", () => {
    const button = renderer.create(<Button text={mockTest} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("Button should be render without text", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("Button should be render with disabled status", () => {
    const button = renderer.create(<Button disabled />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("Button should be render with loader", () => {
    const button = renderer.create(<Button isLoader />).toJSON();
    expect(button).toMatchSnapshot();
  });
});
