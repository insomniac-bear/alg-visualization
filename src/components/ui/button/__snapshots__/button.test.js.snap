// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Button component with snapshot Button should be render with disabled status 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Checking Button component with snapshot Button should be render with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Checking Button component with snapshot Button should be render with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test text
  </p>
</button>
`;

exports[`Checking Button component with snapshot Button should be render without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
